sparse_ho/models/logreg.py

```diff
@@ -380,15 +382,19 @@ def get_hessian(X, y, mask, dense, log_alpha):
             Logarithm of hyperparameter.
         """
         X_m = X[:, mask]
+        n_samples, size_supp = X_m.shape
         a = y * (X_m @ dense)
-        temp = sigma(a) * (1 - sigma(a))
+        grad_sigmoid = sigma(a) * (1 - sigma(a))
         is_sparse = issparse(X)
-        if is_sparse:
-            hessian = csc_matrix(
-                X_m.T.multiply(temp)) @ X_m
-        else:
-            hessian = (X_m.T * temp) @ X_m
-        return hessian
+
+        def mv(v):
+            if is_sparse:
+                return X_m.T @ (grad_sigmoid * (X_m @ v)) / n_samples
```

Comment:
I misunderstood here (the "other example" I'm talking about. But we return the same stuff in both cases, right? If so remove the if

Reply:
done